src/Button/Button.tsx

```diff
@@ -36,32 +38,41 @@ export const Button = ({
       $size={size}
       $variant={variant}
       $hasChildren={hasChildren}
+      $isLoading={isLoading}
       disabled={disabled}
       href={href}
       onClick={onClick}
       {...rest}
     >
-      {startIcon && (
-        <Box mr={hasChildren ? 1 : undefined}>
-          <Icon icon={startIcon} size={sizeIcon[size]} />
-        </Box>
-      )}
-      {hasChildren && (
-        <StyledTypography
-          $color={color}
-          $variant={variant}
-          disabled={disabled}
-          variant={TYPOGRAPHY_VARIANT_MAPPING[size]}
-          forwardedAs="span"
-        >
-          {children}
-        </StyledTypography>
-      )}
-      {endIcon && (
-        <Box ml={hasChildren ? 1 : undefined}>
-          <Icon icon={endIcon} size={sizeIcon[size]} />
-        </Box>
+      {isLoading && (
```

Comment:
Perhaps we can add a ternary

```javascript
{isLoading ? (
          <Spinner size={SPINNER_VARIANT_MAPPING[size]} />
        ) : (
          <>
            {startIcon && (
              <Box mr={hasChildren ? 1 : undefined}>
                <Icon icon={startIcon} size={sizeIcon[size]} />
              </Box>
            )}
            {hasChildren && (
              <StyledTypography
                $color={color}
                $variant={variant}
                disabled={disabled}
                variant={TYPOGRAPHY_VARIANT_MAPPING[size]}
                forwardedAs="span"
              >
                {children}
              </StyledTypography>
            )}
            {endIcon && (
              <Box ml={hasChildren ? 1 : undefined}>
                <Icon icon={endIcon} size={sizeIcon[size]} />
              </Box>
            )}
          </>
        )}
```

Reply:
If we don't render the content of the button and only the spinner, the button will change sizes.

Reply:
Ok, we can review it when you want